states_rebuilder/lib/service/interfaces/i_todo_repository.dart

```diff
@@ -0,0 +1,8 @@
+import '../../domain/entities/todo.dart';
+
+abstract class ITodosRepository {
```

Comment:
Just a general Q about this approach: Why does the Service layer define the Data layer? Doesn't this create a bit of a circular dependency? The Repository Layer depends on the Service layer, and the Service layer depends on the Repository?

Why isn't this interface defined in the repository layer and then the service layer can just import that without the circular dep?

Reply:
The architecture consists of something like onion layers, the innermost layer is the domain layer, the middle layer is the service layer and the outer layer consists : the user interface  UI, repository. 

The dependencies can only point inwards. Nothing in an inner layer can know anything at all about something in an outer layer. In particular, the name of something declared in an outer circle must not be mentioned by the code in the inner circle.

For this reason, the service layer should never import anything from the repository. The service layer define `ITodosRepository`  and decouple the app from its concrete implementation in the repository.

Reply:
Ah, gotcha. I feel like I look at the layers in a slightly different way, but this definitely adheres more to the proper "Clean Architecture" philosophy. Thanks!